connector/connect/common/src/main/protobuf/spark/connect/expressions.proto

```diff
@@ -167,4 +169,26 @@ message Expression {
     // (Optional) Alias metadata expressed as a JSON map.
     optional string metadata = 3;
   }
+
+  message LambdaFunction {
+    // (Required) The lambda function.
+    Expression function = 1;
+
+    // (Required) Function arguments. Must not be empty.
+    //
+    // All arguments must be UnresolvedNamedLambdaVariables.
+    repeated Expression arguments = 2;
+
+    // (Required) A lambda function can be hidden when a user wants to
+    // process an completely independent expression in a 'HigherOrderFunction',
+    // the lambda function and its variables are then only used for internal
+    // bookkeeping within the higher order function.
+    bool hidden = 3;
```

Comment:
Can we see if we can do without this one? This is very much an implementation detail.

Reply:
will remove it.